adafruit_displayio_layout/widgets/widget.py

```diff
@@ -0,0 +1,164 @@
+# SPDX-FileCopyrightText: 2021 Kevin Matocha
+#
+# SPDX-License-Identifier: MIT
+"""
+
+`widget`
+================================================================================
+CircuitPython GUI Widget Class for visual elements
+
+* Author(s): Kevin Matocha
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+
+"""
+
+import displayio
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_DisplayIO_Layout.git"
+
+
+class Widget(displayio.Group):
+    """
+    A Widget class definition for graphical display elements.
+
+    :param int x: pixel position
+    :param int y: pixel position
+    :param int width: width of the switch in pixels, set to ``None`` to auto-size
+     relative to the height
+    :param int height: height of the switch in pixels
+    :param str name: name of the switch
+    :param float anchor_point: (X,Y) values from 0.0 to 1.0 to define the anchor
+     point relative to the switch bounding box
+    :param int anchored_position: (x,y) pixel value for the location
+     of the anchor_point
+
+    """
+
+    def __init__(
+        self,
+        width=None,
+        height=None,
+        name="",
+        anchor_point=None,
+        anchored_position=None,
+        bounding_box=None,  # pixel extent of the widget [x0, y0, width, height]
```

Comment:
Would anything break if someone passed in a `bounding_box` that was different than the x, y, width, and height that they also pass in?

Reply:
Thanks for the feedback.  I decided to get rid of `bounding_box` as an input parameter.  It's more confusing than useful.